### DISCUSSION TITLE
Aave lending pool address change

### DISCUSSION THREAD

**Discussion by PratyushNag** [smartcontractkit/full-blockchain-solidity-course-py#1603] [Q&A] [answered]
In the video Patrick shows that the lending pool address keep changing, therefore we use an AddressProvider to fetch the address for the lending pool contract.

 My question being this, why is it that the address keeps changing, I've not encountered anything like this before.
 and if the contract is able to continually change addresses, how do you accomplish that? is there only a certain set of addresses it keeps switching about, or is it totally different?

I'm having a hard time trying to understand this one particular thing. I tried searching online but I couldn't find anything, apart from the Aave documentation (which also doesn't answer my question as to why and how).
I cant help but get obsessed over small things like this, and it would be very helpful if someone was able to answer.

**Comment by cromewar** [comment, marked as answer]
Hello @PratyushNag 
Really nice question, the reason is because AAVE uses upgradeable contracts, a concept you'll go trough on lesson 12 so the addresses of the lending pool contract will always keep changing, I suggest you just to keep going and in lesson 12 you will understand how upgrades work and how you can accomplish that.
Cheers.

**Reply by PratyushNag** [reply]
Thanks a lot @cromewar
